core/src/main/java/com/linecorp/armeria/server/auth/HttpAuthServiceBuilder.java

```diff
@@ -116,22 +132,52 @@ public HttpAuthServiceBuilder addOAuth2(Authorizer<? super OAuth2Token> authoriz
             }
             return authorizer.authorize(ctx, token);
         };
-        authorizers.add(requestAuthorizer);
+        add(requestAuthorizer);
+        return this;
+    }
+
+    /**
+     * Sets the {@link AuthSuccessHandler} which handles successfully authorized requests.
+     * By default, the request will be delegated to the next {@link Service}.
+     */
+    public HttpAuthServiceBuilder onSuccess(AuthSuccessHandler<HttpRequest, HttpResponse> successHandler) {
+        this.successHandler = requireNonNull(successHandler, "successHandler");
+        return this;
+    }
+
+    /**
+     * Sets the {@link AuthFailureHandler} which handles the requests with failed authorization.
+     * By default, the cause of authorization failure is logged at WARN level (if any) and a
```

Comment:
`an exception thrown during authorization is logged at...`

I guess `result == false` is also an authorization failure but we don't log anything.

Reply:
Thanks for suggestion. Fixed.